packages/admin/admin/src/statusBadge/StatusBadge.tsx

```diff
@@ -0,0 +1,69 @@
+import { CheckRounded, ExclamationmarkRounded, HyphenRounded } from "@comet/admin-icons";
+import { Box, ComponentsOverrides, IconProps, Theme } from "@mui/material";
+import { useTheme } from "@mui/material/styles";
+import { WithStyles, withStyles } from "@mui/styles";
+import * as React from "react";
+
+import { StatusBadgeClassKey, styles } from "./StatusBadge.styles";
+
+export type Status = "success" | "error" | "warning";
+
+const defaultIcons: { [K in Status]: React.FunctionComponentElement<IconProps> } = {
+    success: <CheckRounded fontSize="inherit" />,
+    error: <ExclamationmarkRounded fontSize="inherit" />,
+    warning: <HyphenRounded fontSize="inherit" />,
+};
+
+export interface StatusBadgeProps {
+    status: Status;
+    className?: string;
+    showStatusIcon?: boolean;
+    statusIcon?: React.FunctionComponentElement<IconProps> | undefined;
```

Comment:
I agree that we can remove `showStatusIcon` but the other one is needed imo. If the user wants a custom icon then it's needed. I mean we could use one status prop but as an object with `type` and `icon`

Reply:
There are two versions of the status: with and without icon. We need to support both.

My opinion: Keep `showStatusIcon` and omit `statusIcon`. We have three predefined statuses with three predefined colors so we can also have three predefined icons.

Otherwise we have to make the whole component more flexible. But here I'm more pro ease-of-use vs flexibility.